src/main/java/com/minecolonies/coremod/blocks/BlockPaperwall.java

```diff
@@ -0,0 +1,282 @@
+            new AxisAlignedBB(0.0D, 0.0D, 0.4375D, 1.0D, 1.0D, 1.0D),
+            new AxisAlignedBB(0.4375D, 0.0D, 0.0D, 1.0D, 1.0D, 0.5625D),
+            new AxisAlignedBB(0.4375D, 0.0D, 0.0D, 1.0D, 1.0D, 1.0D),
+            new AxisAlignedBB(0.0D, 0.0D, 0.0D, 1.0D, 1.0D, 0.5625D),
+            new AxisAlignedBB(0.0D, 0.0D, 0.0D, 1.0D, 1.0D, 1.0D)};
+
+
+    /**
+     * The hardness this block has.
+     */
+    private static final float BLOCK_HARDNESS = 3F;
+
+    /**
+     * This blocks name.
+     */
+    private static final String BLOCK_NAME = "blockPaperwall";
+
+    /**
+     * The resistance this block has.
+     */
+    private static final float RESISTANCE = 1F;
+
+    BlockPaperwall()
+    {
+        super(Material.WOOD);
+        initBlock();
+    }
+
+    private void initBlock()
+    {
+        setRegistryName(BLOCK_NAME);
+        setUnlocalizedName(String.format("%s.%s", Constants.MOD_ID.toLowerCase(), BLOCK_NAME));
+        setCreativeTab(ModCreativeTabs.MINECOLONIES);
+        GameRegistry.register(this);
+        GameRegistry.register((new ItemBlock(this)).setRegistryName(this.getRegistryName()));
+        setHardness(BLOCK_HARDNESS);
+        setResistance(RESISTANCE);
+    }
+
+    public void addCollisionBoxToList(@NotNull IBlockState state,
+                                      @NotNull final World worldIn,
+                                      @NotNull final BlockPos pos,
+                                      @NotNull final AxisAlignedBB entityBox,
+                                      @NotNull final List<AxisAlignedBB> collidingBoxes,
+                                      @Nullable final Entity entityIn)
+    {
+        state = this.getActualState(state, worldIn, pos);
+        addCollisionBoxToList(pos, entityBox, collidingBoxes, AABB_BY_INDEX[0]);
+
+        if (state.getValue(NORTH))
+        {
+            addCollisionBoxToList(pos, entityBox, collidingBoxes, AABB_BY_INDEX[getBoundingBoxIndex(EnumFacing.NORTH)]);
+        }
+
+        if (state.getValue(SOUTH))
+        {
+            addCollisionBoxToList(pos, entityBox, collidingBoxes, AABB_BY_INDEX[getBoundingBoxIndex(EnumFacing.SOUTH)]);
+        }
+
+        if (state.getValue(EAST))
+        {
+            addCollisionBoxToList(pos, entityBox, collidingBoxes, AABB_BY_INDEX[getBoundingBoxIndex(EnumFacing.EAST)]);
+        }
+
+        if (state.getValue(WEST))
+        {
+            addCollisionBoxToList(pos, entityBox, collidingBoxes, AABB_BY_INDEX[getBoundingBoxIndex(EnumFacing.WEST)]);
+        }
+    }
+    private static int getBoundingBoxIndex(EnumFacing SIDE)
+    {
+        return 1 << SIDE.getHorizontalIndex();
+    }
+
+    @NotNull
+    public AxisAlignedBB getBoundingBox(@NotNull IBlockState state,
+                                        @NotNull IBlockAccess source,
+                                        @NotNull final BlockPos pos)
+    {
+        state = this.getActualState(state, source, pos);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Introduce a new variable instead of reusing the parameter "state". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1226)